### DISCUSSION TITLE
Put data on device from another process

### DISCUSSION THREAD

**Discussion by thomashirtz** [jax-ml/jax#10436] [General] [answered]
Hello,
I have a neural network that is used to create inference. I would like to maximize the GPU utilization time, however in my case it is putting the data that needs to be processed on the GPU that is taking the most time (data_transfer_time >> training_step_time). Would it be possible to put the batch of data from other process(es) and use the data in the main learner process ?

Thanks!

**Comment by YouJiacheng** [comment, marked as answer]
You may be interested in https://github.com/mpi4jax/mpi4jax
